src/System/Taffybar/HostName.hs

```diff
@@ -0,0 +1,51 @@
+
+module System.Taffybar.HostName
+  ( hostNameNew
+  , defaultHostNameConfig
+  , HostNameConfig(..)
+  ) where
+
+import Data.Monoid ((<>))
+import Graphics.UI.Gtk (Widget, toWidget, widgetShowAll)
+import Network.HostName (getHostName)
+
+import System.Taffybar.Widgets.PollingLabel (pollingLabelNew)
+
+-- | The configuration for the hostname label.
+data HostNameConfig =
+  HostNameConfig
+    { hostNameBackgroundColor :: Maybe String -- ^ Background color. If
+                                              -- 'Nothing', then no background
+                                              -- color is specified. (default
+                                              -- 'Nothing')
+    , hostNameForegroundColor :: String -- ^ Foreground color. (default @\"red\"@)
+    }
+
+defaultHostNameConfig :: HostNameConfig
+defaultHostNameConfig =
+  HostNameConfig
+    { hostNameBackgroundColor = Nothing
+    , hostNameForegroundColor = "red"
+    }
+
+hostNameNew :: HostNameConfig -> IO Widget
+hostNameNew cfg = do
+  label  <- pollingLabelNew "" (60 * 60 * 24) $ getHostNameString cfg
+  widgetShowAll label
+  return $ toWidget label
+
+getHostNameString :: HostNameConfig -> IO String
+getHostNameString cfg = do
+  hostName <- getHostName
+  let bgColorFmt =
+        case hostNameBackgroundColor cfg of
+          Nothing -> ""
+          Just bgColor -> "bgcolor='" <> bgColor <> "' "
+  return $
+    "<span fgcolor='" <>
```

Comment:
Would be much better to use theming support imo. I know that things are done this way in a few places in taffybar atm, but it would be nice to have things done in a more uniform way in the future (especially once the migration to gtk3 is done.

Reply:
You're right.  Using the theming support would be better.